gobblin-metastore/src/main/java/gobblin/metastore/nameParser/GuidDatasetUrnStateStoreNameParser.java

```diff
@@ -0,0 +1,158 @@
+
+    @Getter
+    private String datasetUrnNameMapFile;
+  }
+
+  private final StateStoreNameVersion version;
+  @VisibleForTesting
+  protected final BiMap<String, String> sanitizedNameToDatasetURNMap;
+
+  public GuidDatasetUrnStateStoreNameParser(FileSystem fs, Path jobStatestoreRootDir)
+      throws IOException {
+    this.fs = fs;
+    this.sanitizedNameToDatasetURNMap = Maps.synchronizedBiMap(HashBiMap.<String, String>create());
+    this.versionIdentifier = new Path(jobStatestoreRootDir, StateStoreNameVersion.V1.getDatasetUrnNameMapFile());
+    if (this.fs.exists(versionIdentifier)) {
+      this.version = StateStoreNameVersion.V1;
+      try (InputStream in = this.fs.open(versionIdentifier)) {
+        LineReader lineReader = new LineReader(new InputStreamReader(in, Charsets.UTF_8));
+        String shortenName = lineReader.readLine();
+        while (shortenName != null) {
+          String datasetUrn = lineReader.readLine();
+          this.sanitizedNameToDatasetURNMap.put(shortenName, datasetUrn);
+          shortenName = lineReader.readLine();
+        }
+      }
+    } else {
+      this.version = StateStoreNameVersion.V0;
+    }
+  }
+
+  /**
+   * Get datasetUrn for the given state store name.
+   * If the state store name can be found in {@link #sanitizedNameToDatasetURNMap}, the original datasetUrn will be returned.
+   * Otherwise, the state store name will be returned.
+   */
+  @Override
+  public String getDatasetUrnFromStateStoreName(String stateStoreName)
+      throws IOException {
+    if (version == StateStoreNameVersion.V0 || !this.sanitizedNameToDatasetURNMap.containsKey(stateStoreName)) {
+      return stateStoreName;
+    } else {
+      return this.sanitizedNameToDatasetURNMap.get(stateStoreName);
+    }
+  }
+
+  /**
+   * Merge the given {@link Collection} of datasetUrns with {@link #sanitizedNameToDatasetURNMap}, and write the results
+   * to the {@link Path} of {@link #versionIdentifier}, which is a text file.
+   */
+  @Override
+  public void persistDatasetUrns(Collection<String> datasetUrns)
+      throws IOException {
+    for (String datasetUrn : datasetUrns) {
+      String key = Guid.fromStrings(datasetUrn).toString();
+      if (!this.sanitizedNameToDatasetURNMap.containsKey(key)) {
+        this.sanitizedNameToDatasetURNMap.put(key, datasetUrn);
+      }
+    }
+    Path tmpMapFile = new Path(this.versionIdentifier.getParent(), this.versionIdentifier.getName() + TMP_SUFFIX);
+    try (FSDataOutputStream fsout = this.fs.create(tmpMapFile)) {
+      for (String key : this.sanitizedNameToDatasetURNMap.keySet()) {
+        fsout.write(key.getBytes(Charsets.UTF_8));
+        fsout.writeByte('\n');
+        fsout.write(this.sanitizedNameToDatasetURNMap.get(key).getBytes(Charsets.UTF_8));
+        fsout.writeByte('\n');
+      }
+    }
+    // Back up the previous file first, and then rename the new file.
+    if (this.fs.exists(this.versionIdentifier) && !this.fs.rename(this.versionIdentifier,
```

Comment:
It will be cleaned like state backup files using separate retention jobs. Changed `StateStoreCleaner` for identifying these backups. I also updated javadoc there, since we recommend to use Gobblin-retention for these cleanup jobs.